app/views/ThreadMessagesView/index.tsx

```diff
@@ -459,63 +118,56 @@ class ThreadMessagesView extends React.Component<IThreadMessagesViewProps, IThre
 					useRealName,
 					badgeColor
 				}}
-				onPress={this.onThreadPress}
-				toggleFollowThread={this.toggleFollowThread}
+				onPress={onThreadPress}
+				toggleFollowThread={toggleFollowThread}
 			/>
 		);
 	};
 
-	renderContent = () => {
-		const { loading, messages, displayingThreads, currentFilter } = this.state;
-		const { theme } = this.props;
-		if (!messages?.length || !displayingThreads?.length) {
-			let text;
-			if (currentFilter === Filter.Following) {
-				text = I18n.t('No_threads_following');
-			} else if (currentFilter === Filter.Unread) {
-				text = I18n.t('No_threads_unread');
-			} else {
-				text = I18n.t('No_threads');
-			}
-			return <BackgroundContainer text={text} />;
-		}
+	useLayoutEffect(() => {
+		initSubscription();
```

Comment:
You're using hooks the wrong way.
There shouldn't be a `initSubscription`.
`useSubscription` should return you a subscription to be used.

https://react.dev/learn/you-might-not-need-an-effect